keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -1023,26 +1024,125 @@ this enhancement.
 
 #### Enforcement Actions
 
-For phase 1, all violations implicitly result in a `deny` enforcement action.
+For parity with admission webhooks, a validating policy may also emit audit
+annotations and warnings:
 
-For phase 2, we intend to support multiple enforcement actions.
+- [Audit
+  annotations](https://github.com/kubernetes/kubernetes/blob/97bbf07d3f3f20332912ee411fdf75ce84425e28/staging/src/k8s.io/api/admission/v1/types.go#L142)
+  are key/value pairs included in the audit event for an admission request. The
+  audit annotation key supplied by the policy definition author will be prefixed
+  with the name of the `ValidatingAdmissionPolicy` and policy binding, e.g.:
+  `mypolicy.mygroup.example.com/mybinding.mygroup.example.com/<annotation-key>`.
+- [Warnings](https://kubernetes.io/blog/2020/09/03/warnings/#admission-webhooks)
+  are string messages that are returned to API clients. Warning are returned for
+  both requests that are accepted and requests that are rejected.
```

Comment:
> Warning are returned for both requests that are accepted and requests that are rejected.

I'm not sure I understand this... warn and deny are currently mutually exclusive, selected as the enforcement action by the binding, right?

Reply:
Yes, sorry, let me rephrase this (I was trying to explain the capabilities of webhooks for reference here but it doesn't read well). Your understanding is correct. Policy bindings are the only way to select the enforcement action of validations in this proposal.